src/puppetlabs/puppetdb/cli/services.clj

```diff
@@ -76,8 +76,12 @@
 ;; PuppetDB components.
 
 (def mq-addr "vm://localhost?jms.prefetchPolicy.all=1&create=false")
-(def ^:dynamic mq-endpoint "puppetlabs.puppetdb.commands")
-(def send-command! (partial command/enqueue-command! mq-addr mq-endpoint))
+(def mq-endpoint "puppetlabs.puppetdb.commands")
+
+(defn send-command!
```

Comment:
There's a subtle difference between this code and what was there before. Subtle enough to lose a couple of days you could say :-). One issue in setting up two PuppetDB instances in-process is creating separate command queues for each instance. The previous code closed over the value of mq-endpoint when the instance started up. Even if the mq-endpoint was dynamically bound (which didn't work due to the various threads involved), it didn't matter, because it was closed over at startup. The above code uses the value at the time the command is invoked. This gives enough wiggle room to change the queue name after startup, but before the commands are posted.